### PR TITLE
ci: Add Base RevC 2xAEMvA and AEMv8-R FVPs to CI docker image

### PR DESCRIPTION
**Pull request by ithinuel**
solves https://github.com/zephyrproject-rtos/zephyr/issues/43852

### CODE REVIEW

File: Dockerfile.ci [hunk 1]
```diff
@@ -18,6 +18,8 @@ ENV SPARSE_VERSION=$SPARSE_VERSION
 ARG PROTOC_VERSION=21.7
 ENV PROTOC_VERSION=$PROTOC_VERSION
 ARG WGET_ARGS="-q --show-progress --progress=bar:force:noscroll"
+ARG FVP_BASE_REVC_VERSION=11.24_11
+ARG FVP_BASE_AEMV8R_VERSION=11.24_11
```
Comment (stephanosio) [inline Dockerfile.ci:22]: Please add the corresponding `ENV`s for these (see https://github.com/zephyrproject-rtos/docker-image/pull/164).

File: Dockerfile.ci [hunk 2]
```diff
@@ -92,6 +94,19 @@ RUN mkdir -p /opt/toolchains && \
...
+    cd /opt/fvps && \
+    wget ${WGET_ARGS} https://developer.arm.com/-/media/Files/downloads/ecosystem-models/FVP_Base_RevC-2xAEMvA_${FVP_BASE_REVC_VERSION}_Linux64.tgz && \
+    tar xf FVP_Base_RevC-2xAEMvA_${FVP_BASE_REVC_VERSION}_Linux64.tgz && \
+    ln -s /opt/fvps/Base_RevC_AEMvA_pkg/models/Linux64_GCC-9.3/FVP_Base_RevC-2xAEMvA /usr/local/bin && \
+    rm FVP_Base_RevC-2xAEMvA_${FVP_BASE_REVC_VERSION}_Linux64.tgz && \
+    wget ${WGET_ARGS} https://developer.arm.com/-/media/Files/downloads/ecosystem-models/FVP_Base_AEMv8R_${FVP_BASE_AEMV8R_VERSION}_Linux64.tgz && \
+    tar xf FVP_Base_AEMv8R_${FVP_BASE_AEMV8R_VERSION}_Linux64.tgz && \
+    ln -s /opt/fvps/AEMv8R_base_pkg/models/Linux64_GCC-9.3/FVP_BaseR_AEMv8R /usr/local/bin && \
+    rm FVP_Base_AEMv8R_${FVP_BASE_AEMV8R_VERSION}_Linux64.tgz && \
+    apt-get install -y telnet xterm
```
Comment (stephanosio) [inline Dockerfile.ci:108]: Unless the FVP binaries are architecture-agnostic, they should be encapsulated in `if [ "${HOSTTYPE}" = "x86_64" ]; then` (note that this runs for x86 and AArch64 image builds).

Also are there AArch64 host binaries for FVP? If so, the script should be updated to download AArch64 binaries for AArch64 host.

Comment (stephanosio) [inline Dockerfile.ci:108]: Why is this necessary?

**Review by stephanosio — CHANGES_REQUESTED** [review]
As per the offline discussion, could you please add a note about the redistribution of FVP now being allowed in the commit message?

File: Dockerfile.ci (hunk 2 above)
Comment (ithinuel) [reply Dockerfile.ci:108]: There is only for `Base RevC-2xAEMvA`. `Base_AEMv8R` is only available for `Linux64`. To keep it as a all-or-nothing, I’ll gate it all with `if [ "${HOSTTYPE}" = "x86_64" ]; then` if that’s ok for you.

Comment (stephanosio) [reply Dockerfile.ci]: That is ok for now. Is there any plan to release AArch64 host version in the future? Because the new Zephyr CI runners (WIP) are going to be both x86 and AArch64 (jobs can be scheduled onto both x86 and AArch64 nodes) and feature parity across those two architectures would be desired.

**Review by stephanosio — COMMENTED** [review]
> As per the offline discussion, could you please add a note about the redistribution of FVP now being allowed in the commit message?

Could you also address this?

File: Dockerfile.ci (hunk 2 above)
Comment (ithinuel) [reply Dockerfile.ci]: Corstone-300, Corstone-310 and Armv8-A AEM FVPs have beta version for aarch64 hosts, I don't know the plans for the others.

**Comment by stephanosio** [comment]
> Note: this does not include Corstone-300's FVP.

Is this comment outdated or am I missing something here?

File: Dockerfile.ci [hunk 3]
```diff
@@ -21,6 +21,7 @@ ARG FVP_BASE_REVC_VERSION=11.24_11
 ENV FVP_BASE_REVC_VERSION=$FVP_BASE_REVC_VERSION
 ARG FVP_BASE_AEMV8R_VERSION=11.24_11
 ENV FVP_BASE_AEMV8R_VERSION=$FVP_BASE_AEMV8R_VERSION
+ENV ARMFVP_BIN_PATH=/usr/local/bin
```
[the remark this answers is not in the document]
Comment (ithinuel) [reply Dockerfile.ci]: Done 👍

**Comment by ithinuel** [comment]
> Is this comment outdated or am I missing something here?

Oh, yes indeed. I edited the first post :)

**Review by stephanosio — APPROVED** [review]
